src/actionsHandler.ts

```diff
@@ -0,0 +1,232 @@
+import { PrismaClient } from '@prisma/client';
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import { extractHashOrdinal, extractPagination } from './request-params';
+import { paginatedQuery, fromCreatedAtOrdinalCursor, toCreatedAtOrdinalCursor } from './pagination';
+import { respond, handleError, dagTransactionResponse } from './response';
+
+const prisma = new PrismaClient();
+
+
+// const prisma = new PrismaClient({
+//   log: [
+//     {
+//       emit: "event",
+//       level: "query",
+//     },
+//   ],
+// });
+
+// prisma.$on("query", async (e) => {
+//     console.log(`${e.query} ${e.params}`)
+// });
+
+const dagActionsTables =  [
+    "dag_allow_spends",
+    "dag_spend_transactions",
+    "dag_token_locks",
+    "dag_token_unlocks",
+    "dag_fee_transactions"
+  ];
+
+  const metagraphActionsTables = [
+    "metagraph_allow_spends",
+    "metagraph_spend_transactions",
+    "metagraph_token_locks",
+    "metagraph_token_unlocks",
+    "metagraph_fee_transactions"
+  ];
+
+  const currencyId = (transaction) => (transaction.metagraph_token_lock?.metagraph_id ?? 
+    transaction.metagraph_token_unlock?.metagraph_id ?? 
+    transaction.metagraph_allow_spend?.metagraph_id ?? 
+    transaction.metagraph_spend_transaction?.metagraph_id ??
+    transaction.metagraph_fee_transaction?.metagraph_id)
+  
+  const actionResponse = (transaction) => ({
+    type: transaction.table_name,
+    currencyId: currencyId(transaction),
+    hash: transaction.hash,
+    amount: transaction.amount,
+    source: transaction.source_addr,
+    destination: transaction.destination_addr,
+    unlockEpoch: transaction.last_valid_epoch_progress ?? transaction.unlock_epoch,
+    parentHash: transaction.lock_reference_hash ?? transaction.allow_spend_ref,
+    timestamp: transaction.created_at
+  });
+
+export const actionsResponse = (ts) => ts.map(actionResponse);;
+
+export const actions = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  return await paginatedQuery(
+    extractPagination(event),
+    toCreatedAtOrdinalCursor,
+    fromCreatedAtOrdinalCursor,
+    { 
+      where: { table_name: { in: [...dagActionsTables, ...metagraphActionsTables] } }, 
```

Comment:
We don't have any endpoints that spans across metagraph and DAG actions. Is this supposed to be the DAG actions endpoint (/actions)?